pkg/services/extsvcauth/extsvcaccounts/service.go

```diff
@@ -4,24 +4,32 @@ import (
 	"context"
 	"errors"
 
+	"github.com/grafana/grafana/pkg/components/satokengen"
+	"github.com/grafana/grafana/pkg/infra/db"
 	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/infra/slugify"
 	"github.com/grafana/grafana/pkg/models/roletype"
 	ac "github.com/grafana/grafana/pkg/services/accesscontrol"
 	"github.com/grafana/grafana/pkg/services/extsvcauth"
+	"github.com/grafana/grafana/pkg/services/secrets"
+	"github.com/grafana/grafana/pkg/services/secrets/kvstore"
 	sa "github.com/grafana/grafana/pkg/services/serviceaccounts"
 )
 
 type ExtSvcAccountsService struct {
-	acSvc  ac.Service
-	logger log.Logger
-	saSvc  sa.Service
+	acSvc    ac.Service
+	logger   log.Logger
+	saSvc    sa.Service
+	skvStore kvstore.SecretsKVStore
 }
 
-func ProvideExtSvcAccountsService(acSvc ac.Service, saSvc sa.Service) *ExtSvcAccountsService {
+func ProvideExtSvcAccountsService(acSvc ac.Service, saSvc sa.Service, db db.DB, secretsSvc secrets.Service) *ExtSvcAccountsService {
+	logger := log.New("serviceauth.extsvcaccounts")
 	return &ExtSvcAccountsService{
-		acSvc:  acSvc,
-		logger: log.New("serviceauth.extsvcaccounts"),
-		saSvc:  saSvc,
+		acSvc:    acSvc,
+		logger:   logger,
+		saSvc:    saSvc,
+		skvStore: kvstore.NewSQLSecretsKVStore(db, secretsSvc, logger), // Using SQL store to avoid a cyclic dependency
```

Comment:
I'm not entirely sure I'm allowed to us the SQL store straight away. Need to double check with the plugin squad :slightly_smiling_face:

Reply:
does not sound wrong to me but better double check with the @grafana/backend-platform who owns that.

Reply:
This looks perfectly reasonable to me! We've certainly used this pattern before in places where we could not work around dependency cycles; sometimes it's the only pragmatic option. That's especially true in this case where the kvstore service directly calls the underlying store for all its methods with no extra business logic.

Reply:
Even though codeowner is grafana backend think @grafana/grafana-operator-experience-squad own the unified secrets feature.

Would it help if `kvstore.SecretsKVStore` interface was defined outside of the kvstore package and then you could let wire provide the `SecretsKVStore` dependency?

Reply:
One potential problem with this might be that you take a shortcut around the kvstore and its cache and in addition secrets manager plugin would not be supported if not mistaken

Reply:
> Would it help if `kvstore.SecretsKVStore` interface was defined outside of the kvstore package and then you could let wire provide the `SecretsKVStore` dependency?

Not sure I follow. :slightly_smiling_face: If I understood correctly, the problem lies in the fact that the `SecretKVStore` interface could be provided by a grpc plugin, and since this service aims to set-up credentials for plugins, we have a cyclic dependency injection. 

> One potential problem with this might be that you take a shortcut around the kvstore and its cache

Cache wise, I think this is fine, since this reading/saving is done once on startup, the cache is not necessary.

> in addition secrets manager plugin would not be supported if not mistaken

This is the part I was more concerned about: circumventing the secrets manager plugin.

Reply:
> Edit: Ah I guess there is a confusion 🙂 I'm talking about a cyclic dependency injection problem not an cyclic import problem.

Ohhh I see sorry :cry: Now I remember we've had this problem earlier. It's definitely something we need to figure out moving forward if plugins are supposed to provide core functionality. I'm going to discuss it with my team. 

I guess @grafana/grafana-operator-experience-squad has the verdict if this is a viable way forward for now

Reply:
I got a green light from https://github.com/orgs/grafana/teams/grafana-operator-experience-squad :heavy_check_mark: